backend/Application/Aggregates/Contract/Dto/TransactionResultEventDto.cs

```diff
@@ -0,0 +1,26 @@
+using Application.Api.GraphQL.Accounts;
+using Application.Api.GraphQL.Transactions;
+
+namespace Application.Aggregates.Contract.Dto;
+
+/// <summary>
+/// Used as a Dto when <see cref="TransactionResultEvent"/> from a block height is queried.
+/// </summary>
+public class TransactionResultEventDto
+{
+    public int BlockHeight { get; init; }
+    public TransactionTypeUnion TransactionType { get; init; }
+    public AccountAddress? TransactionSender { get; init; }
+    public string TransactionHash { get; init; }
+    public uint TransactionIndex { get; init; }
+    public uint TransactionEventIndex { get; init; }
+    public TransactionResultEvent Event { get; init; }
+    
+    /// <summary>
+    /// Needed for Dapper
+    /// </summary>
+#pragma warning disable CS8618
```

Comment:
Change to follow what EF core propose in case of C# 10
https://learn.microsoft.com/en-us/ef/core/miscellaneous/nullable-reference-types#non-nullable-properties-and-initialization